spec/truffle/interop/java_spec.rb

```diff
@@ -8,9 +8,23 @@
 
 require_relative '../../ruby/spec_helper'
 
+guard -> { Truffle.native? } do
+  describe "Java" do
+    
+    it "is not defined" do
+      defined?(Java).should be_nil
+    end
```

Comment:
What happens if `java` is required on native?

It seems to still define it even though it crashes later (so good enough for this check):
```
$ graalvm-0.32/bin/ruby --native -e 'begin;require "java";ensure;p Java; end'
Java
graalvm-0.32/jre/languages/ruby/lib/truffle/java/java_utilities.rb:53:in `java_class_by_name_and_loader': Not available on SVM. (RuntimeError)
	from graalvm-0.32/jre/languages/ruby/lib/truffle/java/java_utilities.rb:53:in `java_class_by_name'
```
Probably best to have an early check before defining the Java module, raising a LoadError or so. cc @aardvark179

Reply:
Good idea, but the JRuby-compatible interop is a very minor, untested feature. Not fixing here.

Reply:
@aardvark179 Could you add this check and also add a simple JRuby-compatible Java interop test? Currently the only test is `no_extra_output.sh` which only tests that `require "java"` works.